src/app/(protected)/workspace/[namespace]/settings/general/_components/general-form.tsx

```diff
@@ -20,28 +20,34 @@ import DeleteWorkspace from "./delete-workspace";
 import { api } from "~/trpc/react";
 import { useRouter } from "next/navigation";
 import { toast } from "sonner";
-import { Icons } from "~/components/icons";
 import { useWorkspace } from "../../../workspace-provider";
 import { env } from "~/env";
+import { handleTrpcError } from "~/lib/utils";
 import { updateWorkspace } from "~/types/workspace";
+import { workspace } from "~/schemas/public/Workspace";
 
 type Props = {
-  workspaceId: string;
+  workspace: z.infer<typeof workspace>;
```

Comment:
No need to infer here haha, there is a Workspace type in that file

Reply:
Didn't notice lol. Thanks